sql/resource_widget.go

```diff
@@ -268,6 +268,16 @@ func ResourceWidget() *schema.Resource {
 			m["visualization_id"].DiffSuppressFunc = func(_, old, new string, d *schema.ResourceData) bool {
 				return extractVisualizationID(old) == extractVisualizationID(new)
 			}
+			p, err := common.SchemaPath(m, "position", "pos_x")
+			if err == nil {
+				p.Optional = true
+				p.Required = false
+			}
+			p, err = common.SchemaPath(m, "position", "pos_y")
+			if err == nil {
+				p.Optional = true
+				p.Required = false
+			}
```

Comment:
Both of these are required in the payload, AFAIK. Why default to 0?

Reply:
Maybe because `0` is the same as unset for non-pointer values?

Reply:
yes, but they should be ignored only if they are declared as `omitempty`, and this is not a case